homeassistant/components/sensor/waze_travel_time.py

```diff
@@ -98,54 +130,97 @@ def icon(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes of the last update."""
-        return {
-            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
-            ATTR_DISTANCE: round(self._state['distance']),
-            ATTR_ROUTE: self._state['route'],
-        }
-
-    def update(self):
-        """Fetch new state data for the sensor."""
-        try:
-            self.waze_data.update()
-            self._state = self.waze_data.data
-        except KeyError:
-            _LOGGER.error("Error retrieving data from server")
-
-
-class WazeRouteData(object):
-    """Get data from Waze."""
-
-    def __init__(self, origin, destination, region, incl_filter, excl_filter):
-        """Initialize the data object."""
-        self._destination = destination
-        self._origin = origin
-        self._region = region
-        self._incl_filter = incl_filter
-        self._excl_filter = excl_filter
-        self.data = {}
+        if self._state is None:
+            return None
+
+        res = {ATTR_ATTRIBUTION: CONF_ATTRIBUTION}
+        if 'duration' in self._state:
+            res[ATTR_DURATION] = self._state['duration']
+        if 'distance' in self._state:
+            res[ATTR_DISTANCE] = self._state['distance']
+        if 'route' in self._state:
+            res[ATTR_ROUTE] = self._state['route']
+        return res
+
+    def _get_location_from_entity(self, entity_id):
+        """Get the location from the entity state or attributes."""
+        state = self.hass.states.get(entity_id)
+
+        if state is None:
+            _LOGGER.error('Unable to find entity %s', entity_id)
```

Comment:
We prefer double quotes for logging messages.

https://developers.home-assistant.io/docs/en/development_guidelines.html#quotes

https://developers.home-assistant.io/docs/en/development_guidelines.html#log-messages